src/Compilers/CSharp/Portable/Emitter/Model/PEModuleBuilder.cs

```diff
@@ -890,6 +891,23 @@ protected virtual Cci.IModuleReference TranslateModule(ModuleSymbol module, Diag
             return namedTypeSymbol;
         }
 
+        private void CheckTupleUnderlying(NamedTypeSymbol namedTypeSymbol, SyntaxNode syntaxNodeOpt, DiagnosticBag diagnostics)
+        {
+            // check that underlying type of a ValueTuple is indeed a value type (or error)
+            // this should never happen, in theory,
+            // but if it does happen we should make it a failure.
+            // NOTE: declaredBase could be null for interfaces
+            var declaredBase = namedTypeSymbol.BaseTypeNoUseSiteDiagnostics;
+            if (declaredBase?.SpecialType != SpecialType.System_ValueType && declaredBase?.IsErrorType() != false)
+            {
+                // Try to decrease noise by not complaining about the same type over and over again.
+                if (_reportedErrorTypesMap.Add(namedTypeSymbol))
+                {
+                    diagnostics.Add(new CSDiagnostic(new CSDiagnosticInfo(ErrorCode.ERR_PredefinedValueTupleTypeMustBeStruct, namedTypeSymbol.MetadataName), syntaxNodeOpt == null ? NoLocation.Singleton : syntaxNodeOpt.Location));
```

Comment:
A tuple of cardinality >= 8 is used and VT`8 is not a structure, or "short" VT is not a structure, or both are not structures.